[Bug]: Unknown snapshot issue found in soak tests

### Problem encountered

On the system test branch `openining_auction_fix` a new test was added. During a full system test run with that new tests, the snapshot-soak job picked up failing snapshots that were created during that test:
```
[2023-06-07T18:37:34.170Z] percentage of working snapshots: 99.3801652892562% (481/484)

[2023-06-07T18:37:34.170Z] [1320, 1298, 1276]

[2023-06-07T18:37:34.170Z] ........................................................................................................................................................................................................................................................................................................................................................................................................................................FFF.........................................................
```

Its currently unknown whether the snapshot issue is one that:
- already exists in mainnet, and was just uncovered by the new tests
- has been introduced into the code since mainnet (and may or may not be in the upcoming patch)

### Observed behaviour

For a particular combination of core state, a snapshot is taken that does not work when used to restore a node.

### Expected behaviour

For all combinations of core state, any snapshots taken work when used to restore a node.

### Steps to reproduce

```bash
1. Download the artefacts from this CI run: https://jenkins.ops.vega.xyz/blue/organizations/jenkins/common%2Fsystem-tests-wrapper/detail/system-tests-wrapper/73924/pipeline 
2. checkout this script: https://github.com/vegaprotocol/jenkins-shared-library/blob/main/resources/bin/pv-snapshot-all
3. run it with `pv-snapshot-all -r`
4. run it again with `pv-snapshot-all -b 1320`
```


### Software version

core develop

### Failing test

_No response_

### Jenkins run

https://jenkins.ops.vega.xyz/blue/organizations/jenkins/common%2Fsnapshot-soak-tests/detail/snapshot-soak-tests/6510/pipeline

### Configuration used

_No response_

### Relevant log output

```shell
panic: block.AppHash does not match AppHash after replay. Got 3BB94C087312F5B6A423AD02331B787D0AFC738EC753A71DE15D797928895362, expected 6191B151A9269515F4FE4B1FF921BFF0CF95472C46EED98DD275AF938B460472.

Block: Block{
  Header{
    Version:        {11 1}
    ChainID:        testnet-001
    Height:         1325
    Time:           2023-06-08 09:36:54.043114736 +0000 UTC
    LastBlockID:    6D5343F2635488794786F476606E1D68C7BF2CFDEC88009F714635B52D55CDB3:1:9394F8ABEA5B
    LastCommit:     70B99635BD34DFB634003D3F5700B832BB8526494E36AA231E53C83478DF86B6
    Data:           1408D47AD51C33125AE95FA5F9C965E6A5AB78D7F8E4953AB4ECB9212A34A3BE
    Validators:     7C0755828182A6CE6AC10041C01DFAA43895491D4D1275173141F491238CB140
    NextValidators: 7C0755828182A6CE6AC10041C01DFAA43895491D4D1275173141F491238CB140
    App:            6191B151A9269515F4FE4B1FF921BFF0CF95472C46EED98DD275AF938B460472
    Consensus:      C8CD39A92B22597AC33565A59990B4D1E440B15621B5E932D1D5B39A8F45E9DB
    Results:        E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
    Evidence:       E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
    Proposer:       BACF85663DB4E44285016CBA4CE80F4D1F123FBD
  }#05550CEA4C703C83DDCDC2C1393210E9CCEA9D8B65DF774404BBC41D90A65E5F
  Data{
    7394FAE79E54BC675C66ABAABF9443EF39A6E9DEF2889347BA72B9A7B9082A29 (458 bytes)
  }#1408D47AD51C33125AE95FA5F9C965E6A5AB78D7F8E4953AB4ECB9212A34A3BE
  EvidenceData{
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -51,6 +51,7 @@
 - [8225](https://github.com/vegaprotocol/vega/issues/8225) - Better error handling in `ListEntities`
 - [8222](https://github.com/vegaprotocol/vega/issues/8222) - `EstimatePositions` does not correctly validate data.
 - [8266](https://github.com/vegaprotocol/vega/issues/8266) - Fix HTTPS with `autocert`.
+- [8471](https://github.com/vegaprotocol/vega/issues/8471) - Restore network parameters from snapshot without validation to avoid order dependence.
 - [8290](https://github.com/vegaprotocol/vega/issues/8290) - Calling network history `API` without network history enabled caused panics in data node.
 - [8299](https://github.com/vegaprotocol/vega/issues/8299) - Fix listing of internal data sources in GraphQL.
 - [8279](https://github.com/vegaprotocol/vega/issues/8279) - Avoid overriding a map entry while iterating on it, on the wallet connection manager.

--- a/core/netparams/snapshot.go
+++ b/core/netparams/snapshot.go
@@ -131,7 +131,9 @@ func (s *Store) LoadState(ctx context.Context, pl *types.Payload) ([]types.State
 	}
 
 	for _, kv := range np.NetParams.Params {
-		s.Update(ctx, kv.Key, kv.Value)
+		if err := s.UpdateOptionalValidation(ctx, kv.Key, kv.Value, false); err != nil {
+			return nil, err
+		}
 	}
 
 	// Now they have been loaded, dispatch the changes so that the other engines pick them up

--- a/core/netparams/snapshot_test.go
+++ b/core/netparams/snapshot_test.go
@@ -1,0 +1,65 @@
+// Copyright (c) 2022 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE.VEGA file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package netparams_test
+
+import (
+	"context"
+	"testing"
+
+	"code.vegaprotocol.io/vega/core/netparams"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/proto"
+	snapshot "code.vegaprotocol.io/vega/protos/vega/snapshot/v1"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSnapshotRestoreDependentNetparams(t *testing.T) {
+	netp := getTestNetParams(t)
+	defer netp.ctrl.Finish()
+	ctx := context.Background()
+
+	netp.broker.EXPECT().Send(gomock.Any()).AnyTimes()
+
+	// get the original default value
+	err := netp.Update(
+		context.Background(), netparams.MarketAuctionMinimumDuration, "1s")
+	assert.NoError(t, err)
+
+	// now change max
+	err = netp.Update(
+		context.Background(), netparams.MarketAuctionMaximumDuration, "10s")
+	assert.NoError(t, err)
+
+	// now snapshot restore
+	data, _, err := netp.GetState("all")
+	require.NoError(t, err)
+
+	snap := &snapshot.Payload{}
+	err = proto.Unmarshal(data, snap)
+	require.Nil(t, err)
+
+	snapNetp := getTestNetParams(t)
+	snapNetp.broker.EXPECT().Send(gomock.Any()).AnyTimes()
+	_, err = snapNetp.LoadState(ctx, types.PayloadFromProto(snap))
+	require.NoError(t, err)
+
+	v1, err := snapNetp.Get(netparams.MarketAuctionMaximumDuration)
+	require.NoError(t, err)
+	v2, err := netp.Get(netparams.MarketAuctionMaximumDuration)
+	require.NoError(t, err)
+
+	require.Equal(t, v1, v2)
+}

